examples/support-vector-machine/script.jl

```diff
@@ -17,16 +17,15 @@ Random.seed!(1234);
 # ## Generate half-moon dataset
 
 # Number of samples per class:
-nin = nout = 50;
+n1 = n2 = 50;
 
 # We generate data based on SciKit-Learn's sklearn.datasets.make_moons function:
 
-class1x = cos.(range(0, π; length=nout))
-class1y = sin.(range(0, π; length=nout))
-class2x = 1 .- cos.(range(0, π; length=nin))
-class2y = 1 .- sin.(range(0, π; length=nin)) .- 0.5
-X = hcat(vcat(class1x, class2x), vcat(class1y, class2y))
-X .+= 0.1randn(size(X))
+angle1 = range(0, π; length=n1)
+angle2 = range(0, π; length=n2)
+X1 = [cos.(angle1) sin.(angle1)] .+ 0.1randn(n1, 2)
+X2 = [1 .- cos.(angle2) 1 .- sin.(angle2) .- 0.5] .+ 0.1randn(n2, 2)
```

Comment:
`0.1randn(n1, 2)` creates two arrays that can be avoided:
```suggestion
X1 = [cos.(angle1) sin.(angle1)] .+ 0.1 .* randn.()
X2 = [1 .- cos.(angle2) 1 .- sin.(angle2) .- 0.5] .+ 0.1 .* randn.()
```
Alternatively, with one allocation:
```suggestion
X1 = [cos.(angle1) sin.(angle1)] .+ 0.1 .* randn(n1, 2)
X2 = [1 .- cos.(angle2) 1 .- sin.(angle2) .- 0.5] .+ 0.1 .* randn(n2, 2)
```

Reply:
In this case, the number of allocations makes no practical difference... but not having to pass the shape is neat! I didn't know `randn.()` works like that, I would have thought it only shifts everything by the same amount.

Reply:
Damn! This is nice! I never new about  `x .+ randn.()` is it working because of broadcast fusion?